pkg/codegen/python/gen.go

```diff
@@ -791,9 +790,6 @@ func (mod *modContext) importObjectType(t *schema.ObjectType, input bool) string
 	}
 
 	importPath := mod.getRelImportFromRoot()
-	if mod.pkg.Name() != parts[0] {
```

Comment:
If we're not using the package part of the token to determine if it's a local or external import do we need to be checking something else?

Although looking at existing python tests (like the external-enum test) this looks like it's borked anyway because we're getting type annotations to things that don't exist. I guess the python codegen tests don't actually run mypy to check the typing annotations.

Reply:
> If we're not using the package part of the token to determine if it's a local or external import do we need to be checking something else?

We do the check earlier in the method:

https://github.com/pulumi/pulumi/blob/44f20644839943e0a9bc300213ee98d3e9aa6112/pkg/codegen/python/gen.go#L776-L778

What I'm removing seems to be leftover from a previous attempt to support this. But I did it very late at night so I'm sure I messed some things up. Need to look over the test results.

> Although looking at existing python tests (like the external-enum test) this looks like it's borked anyway because we're getting type annotations to things that don't exist.

Hmm, yeah, `external-enum` test output for Python does look screwed up. I'll take a look as well.

> I guess the python codegen tests don't actually run mypy to check the typing annotations.

We should fix up the compile step to run mypy for these. I'll open a separate issue.

Reply:
Opened https://github.com/pulumi/pulumi/issues/12694 to track adding mypy checking support (I have a WIP, but will do it as a separate change).

Also opened https://github.com/pulumi/pulumi/issues/12695 about referencing enums.